File: packages/shared/src/schemas/session.ts
import { z } from "zod";

// --- Session entity (§6.2) ---

export const Session = z.object({
  id: z.string(),
  hostId: z.string(),
  provider: z.literal("codex"),
  codexSessionId: z.string().nullable(),
  title: z.string(),
  cwd: z.string().nullable(),
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime(),
  lastPreview: z.string().nullable(),
  archivedAt: z.string().datetime().nullable(),
});
export type Session = z.infer<typeof Session>;

File: apps/android/app/src/main/java/app/findeck/mobile/data/model/Session.kt
package app.findeck.mobile.data.model

import kotlinx.serialization.Serializable

/** Mirrors the shared Session schema from packages/shared. */
@Serializable
data class Session(
    val id: String,
    val hostId: String,
    val provider: String = "codex",
    val codexSessionId: String? = null,
    val title: String,
    val cwd: String? = null,
    val createdAt: String,
    val updatedAt: String,
    val lastPreview: String? = null,
    val archivedAt: String? = null,
)

@Serializable
data class ListSessionsResponse(
    val sessions: List<Session>,
)

@Serializable
data class ArchiveSessionsResponse(
    val ok: Boolean,
    val archivedCount: Int,
)

@Serializable
data class UnarchiveSessionsResponse(
    val ok: Boolean,
    val unarchivedCount: Int,
)
